core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -60,8 +61,7 @@ private[spark] class PipedRDD[T: ClassTag](
       printRDDElement: (T, String => Unit) => Unit = null,
       separateWorkingDir: Boolean = false) =
     this(prev, PipedRDD.tokenize(command), envVars, printPipeContext, printRDDElement,
-      separateWorkingDir, 8192)
-
+      separateWorkingDir, 8192, Codec.defaultCharsetCodec.name)
```

Comment:
I think we can remove this whole constructor actually. I don't see it is used anywhere and it's internal. It might help keep this a little simpler, if so. (The comment above it at least looks unrelated.)